var/spack/repos/builtin/packages/dbcsr/package.py

```diff
@@ -45,15 +46,16 @@ class Dbcsr(CMakePackage, CudaPackage, ROCmPackage):
     )
 
     variant("opencl", default=False, description="Enable OpenCL backend")
+    variant("mpi_f08", default=False, description="Use mpi F08 module")
 
     depends_on("blas")
     depends_on("lapack")
     depends_on("mpi", when="+mpi")
     depends_on("libxsmm@1.11:~header-only", when="smm=libxsmm")
 
     depends_on("cmake@3.17:", type="build", when="@2.1:")
-    depends_on("cmake@3.10:", type="build")
-    depends_on("py-fypp", type="build")
+    depends_on("cmake@3.22:", type="build")
```

Comment:
Is this version dependent? This subsumes the constraint above, but I'm guessing you want instead:
```suggestion
    depends_on("cmake@3.10:", type="build")
    depends_on("cmake"3.22:", type="build", when="@2.6:") # maybe 2.6:?
```

Reply:
I did not write the cmake build system so I will assume it the case. I think cmake is backwards compatible so we could remove this dependence versioning altogether. I really doubt anyone will use anything below 2.4 here but who knows.